### PR TITLE
ng-dialog updates

### PR DESCRIPTION
**Pull request by michaelwittwer**
extended the typings of ng-dialog reading the official docs at https://github.com/likeastore/ngDialog

### CODE REVIEW

File: ng-dialog/ng-dialog.d.ts [hunk 1]
```diff
@@ -7,104 +7,250 @@
```
Comment (stephenlautier) [inline ng-dialog/ng-dialog.d.ts]: Can you please change whitespace to tabs as they were

Comment (michaelwittwer) [inline ng-dialog/ng-dialog.d.ts]: updated the formatting

File: ng-dialog/ng-dialog.d.ts [hunk 2]
```diff
@@ -83,6 +136,12 @@ declare module angular.dialog {
 		closeByDocument?: boolean;
 
 		/**
+		 * default : false
+		 */
+		closeByNavigation?: boolean;
```
Comment (stephenlautier) [inline ng-dialog/ng-dialog.d.ts]: Are you sure this one exists? I can't find it within the docs

Comment (stephenlautier) [inline ng-dialog/ng-dialog.d.ts]: Ah but it is within code, https://github.com/likeastore/ngDialog/blob/master/js/ngDialog.js#L48 .. not sure what it does, but if you know you can add some comments to the code :)

File: ng-dialog/ng-dialog.d.ts [hunk 3]
```diff
@@ -41,18 +52,54 @@ declare module angular.dialog {
...
 		 * @param value Any value passed to this function will be attached to the object which resolves on the close promise for this dialog.
 		 * For dialogs opened with the openConfirm() method the value is used as the reject reason.
 		 */
 		closeThisDialog(value?: any): void;
+
+		/**
+		 * Any serializable data that you want to be stored in the controller's dialog scope.
+		 * From version 0.3.6 $scope.ngDialogData keeps references to the objects instead of copying them.
+		 */
+		ngDialogData : {};
```
Comment (stephenlautier) [inline ng-dialog/ng-dialog.d.ts]: I think this should be as `string | {} | any[]; `, to match the `data` provided within the `IDialogOpenOptions`

**Comment by stephenlautier** [comment]
Also make sure you pull before to your branch as there seems to be conflicts

File: ng-dialog/ng-dialog.d.ts (hunk 1 above)
Comment (stephenlautier) [inline ng-dialog/ng-dialog.d.ts]: Thanks, makes it much easier :+1:

**Comment by michaelwittwer** [comment]
hmm i guess i just really messed up the whole thing be pulling from forked repo (should i rebase doing this?) Other than that i changed what you commented on, just gimme feedback if I should start over with a new fork and just copy the relevant info.

**Comment by stephenlautier** [comment]
doesnt matter, if you can fix this branch it will be better tho. if it gets messy just close this PR and recreate with the changes

**Comment by michaelwittwer** [comment]
What do you think about the PR, can you merge it in?

**Comment by stephenlautier** [comment]
There's the CI giving errors, they can't merge it unless you fix it. Locally within your root run `npm test`, once everything is fine I can approve and they will be able to merge

File: ng-dialog/ng-dialog.d.ts [hunk 4]
```diff
@@ -7,10 +7,21 @@
 
 declare module angular.dialog {
 
+	/*
+	 * Everytime ngDialog is opened or closed we're broadcasting three events
+	 * (dispatching events downwards to all child scopes):
+	 *
+	 * for more info see: https://github.com/likeastore/ngDialog#events
+	 */
+	export const EVENT_OPENEND:string = 'ngDialog.opened';
```
Comment (stephenlautier) [inline ng-dialog/ng-dialog.d.ts]: I don't think you are able to do any constants (code) within the d.ts file, only a contract (definition).

File: ng-dialog/ng-dialog.d.ts [hunk 5]
```diff
@@ -41,18 +52,54 @@ declare module angular.dialog {
...
+	}
+
+	interface IDialogOpenConfirmScope extends IDialogOpenScope {
+		/**
+		 * Use this method to close the dialog and resolve the promise that was returned when opening the modal.
+		 *
+		 * The function accepts a single optional parameter which is used as the value of the resolved promise.
+		 * @param {any} [value] - The value with which the promise will resolve
+		 */
+		confirm(value?: any)
```
Comment (stephenlautier) [inline ng-dialog/ng-dialog.d.ts]: You have to specify a return type for the method

**Comment by michaelwittwer** [comment]
Thanks for your inputs. Updated the PR, should all work now.

**Comment by stephenlautier** [comment]
:+1:  looks good, thanks mate!

**Comment by michaelwittwer** [comment]
Thanks. Do you think you can merge it in soon?

**Comment by stephenlautier** [comment]
@michaelwittwer sorry I do not have access to merge, only review. @vvakame usually merges.
@vvakame can you merge, everything is good :+1: . Thanks.

**Comment by vvakame** [comment]
:+1:

**Pull request merged**